docs/polkadot/networking/3-attestation.md

```diff
@@ -1,54 +1,268 @@
-- Chain Heads: `Vec<Hash>` up to arbitrary constant N of viable leaves according to that peer. in practice there will only be 1 or 2 but let's say N=5. None of the chain heads may be descendants of each other or ancestors of any chain heads sent in the prior packet. Doing this check for all chain heads in all prior updates is not tractable, but it's possible this could be left open to a "flip-flop" attack of some kind.
+Let T_l represent the average latency (single trip) between two peers, i.e. RTT is twice this.
 
-**Gossip Messages:**
-- **Candidate:**
-    - Members
-        - Parachain candidate
-        - Validator public key
-        - Signature
-        - Relay Parent
-    - This is an announcement of a parachain candidate to the network. Only one announcement may be made by each validator per relay parent hash. Receiving this message from a peer indicates that the peer is aware of the candidate. Sending this message to a peer will make the peer aware of the statement, and should be done only if there is no prior indication of awareness.
-- **Valid:**
-    - Members
-        - Candidate Hash
-        - Validator public key
-        - Signature
-        - Relay Chain Parent Hash
-    - This is an attestation on the candidate with given hash, that indicates the candidate is both valid and has block data available. May only be sent to peers who are aware of the `Candidate`.
-- **Invalid:**
-    - Members
-        - Candidate Hash
-        - Validator public key
-        - Signature
-        - Relay Chain Parent Hash
-    - This is an attestation against the candidate with given hash. May only be sent to peers who are aware of the `Candidate`.
+Let T_B represent the average time it takes for a distributor to transfer all pieces of a block, i.e. a single piece takes roughly T_B/N. This is only the bandwidth and not latency, so only if T_B/N is negligible then the successful direct transfer of a single piece including acknowledge takes about 1 RTT.
 
-### Direct Routing: sending/ receiving erasure coded pieces
+Let T_b be T_B divided by the number of validators per parachain, i.e. T_b = T_B/(N/C). This represents the average time it takes for a distributor to transfer the pieces of a block if this responsibility was shared equally across all validators of a parachain.
+
+Let T_L represent the standard deviation of the distribution of when validators begin this distribution process for each period.
+
+In practise based on some crude real-world measurements, we guess that T_l is about 150ms, T_b is about 3s, and T_L is about 750ms. In other words, T_b ~= 20 * T_l ~= 4 * T_L.
+
+## High-level requirements
+
+Ensure that ceil(N/3) of the pieces of all C blocks remains available and retrievable for a reasonable amount of time, across the N validators.
+
+## Design considerations
+
+### General aims
+
+A1: Reduced redundancy & latency: pieces should (mostly) only be sent to the nodes that need it, i.e. its main storer, and the secondary checkers.
```

Comment:
Can you elaborate what a *checker* is here? As far as I can tell https://research.web3.foundation/en/latest/polkadot/Availability_and_Validity.html does not use this term.

Reply:
I've added a part describing this to "high-level requirements" (near the top).

We've started using the terms "preliminary checker" and "approval checker" instead of primary vs secondary checker, and I've update this document accordingly.